tests/commands/test_mkdir.py

```diff
@@ -1,147 +1,144 @@
+import pytest
 import subprocess
 from pathlib import Path
 
+from onyo.lib import Repo
 
-def anchored_dir(directory):
+
+directories = ['simple',
+               's p a c e s',
+               's p a/c e s',
+               'overlap/one',
+               'overlap/two',
+               'overlap/three',
+               'r/e/c/u/r/s/i/v/e',
+               'spe\"cial\\char\'actஞers',
+               'very/very/very/deep'
+               ]
+
+
+@pytest.mark.parametrize('directory', directories)
+def test_mkdir(repo: Repo, directory: str) -> None:
     """
-    Returns True if a directory exists and contains an .anchor file.
-    Otherwise it returns False.
+    Test that `onyo mkdir <dir>` creates new directories correctly for different
+    depths and directory names.
     """
-    if Path(directory).is_dir() and Path(directory, '.anchor').is_file():
-        return True
+    ret = subprocess.run(['onyo', 'mkdir', directory], capture_output=True, text=True)
 
-    return False
+    # verify output
+    assert not ret.stdout
+    assert not ret.stderr
+    assert ret.returncode == 0
 
+    # verify folders and anchors exist
+    d = Path(directory)
+    while not d.samefile(repo.root):
```

Comment:
An alternative is to loop over `Path().parents`.

Reply:
Looping over `Path().parents` starts then directly with the parent (and not the Path), which is the most important to check, and all the ways to include it does look worse in my opinion.

So I think still this is the best version to reach that functionality.
If you see a way doing it with `Path().parents` that does this better, let me know, because I did not like the ways I came up using it.

Reply:
Makes sense. :-)